connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/client/JdbiJsonHelper.java

```diff
@@ -0,0 +1,53 @@
+package io.camunda.connector.jdbc.model.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.jdbi.v3.core.qualifier.QualifiedType;
+import org.jdbi.v3.core.result.ResultIterable;
+import org.jdbi.v3.core.result.UnableToProduceResultException;
+import org.jdbi.v3.core.statement.Query;
+import org.jdbi.v3.json.Json;
+
+public class JdbiJsonHelper {
+
+  public static ResultIterable<Map<String, Object>> mapToParsedMap(Query query) {
+    return query.map(
+        (rs, ctx) -> {
+          Map<String, Object> row = new HashMap<>();
+          for (int i = 1; i <= rs.getMetaData().getColumnCount(); i++) {
+            String columnName = rs.getMetaData().getColumnLabel(i);
+            Object value = rs.getObject(i);
+            if (isPotentiallyJsonColumn(rs.getMetaData().getColumnType(i), value)) {
```

Comment:
Did you try using `rs.getMetaData().getColumnTypeName`?

It might be more useful as it returns the
> type name used by the database. If the column type is a user-defined type, then a fully-qualified type name is returned.

Instead of the JDBC type returned by `getColumnType`.

Reply:
Yes, @johnBgood and I have thought about this. We decided to go with the `getColumnType` function, because there we get a List of SQL types that are not depending on the SQL dialect but general for all SQL databases, which is what we want. Some SQL dialects do not have specific JSON datatypes, but use VARCHAR for JSONs. 

If we would use `getColumnTypeName` we need to check for all types of all dialects, which makes the check a lot more complex, without any additional benefit.

Reply:
> If we would use getColumnTypeName we need to check for all types of all dialects, which makes the check a lot more complex, without any additional benefit.

But currently we try to parse simple strings as json just to figure out whether it works or not right?

Reply:
You can see in the test which dialect handels it how:

MSSQL: NVARCHAR
MYSQL: JSON
POSTGRESQL: JSON, JSONB
MARIADB: JSON

Reply:
>  put jsons in VARCHAR columns
Yes, but thats something that needs to be handled in userland from my perspective. We should only cover when json is managed in fields we know that contain json.

Reply:
Otherwise we are just guessing and bruteforce json deserialization on every string (plus the other types that are defined in the method).

Reply:
ok. Thanks for explaining. I adjusted it.